packages/widgetbook/test/src/addons/utils/widget_wrapper.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:widgetbook/src/addons/addon_provider.dart';
+import 'package:widgetbook/src/addons/device_addon/device_provider.dart';
+import 'package:widgetbook/src/addons/device_addon/frame_provider.dart';
+import 'package:widgetbook/src/addons/theme_addon/theme_provider.dart';
+import 'package:widgetbook/src/addons/theme_addon/theme_selection_provider.dart';
+import 'package:widgetbook/widgetbook.dart';
+
+import 'custom_app_theme.dart';
+
+final customTheme = WidgetbookTheme<AppThemeData>(
+  name: 'Blue',
+  data: AppThemeData(color: Colors.blue),
+);
+
+final customTheme2 = WidgetbookTheme<AppThemeData>(
+  name: 'Yellow',
+  data: AppThemeData(color: Colors.yellow),
+);
+
+final devices = [
+  Apple.iPhone11,
+  Apple.iPhone12,
+  const Device.special(
+    name: 'Test',
+    resolution: Resolution(
+      nativeSize: DeviceSize(width: 400, height: 400),
+      scaleFactor: 1,
+    ),
+  ),
+];
+
+final deviceFrameBuilder = DeviceFrameBuilder(
+  devices: devices,
+);
+
+final activeFrameBuilder = WidgetbookFrameBuilder(
+  devices: devices,
+);
+
+extension TestExtensions on Widget {
+  Widget addOnProviderWrapper() => MultiProvider(
+        providers: [
+          ChangeNotifierProvider(
+            key: ValueKey(
+              devices.take(1).first,
+            ),
+            create: (context) => DeviceProvider(
+              devices.take(1).first,
+            ),
+          ),
+          ChangeNotifierProvider(
+            create: (_) => ThemeProvider(customTheme),
+          ),
+          ChangeNotifierProvider(
+            create: (_) => FrameProvider(
+              activeFrameBuilder,
+            ),
+          ),
+          ChangeNotifierProvider(
+            create: (_) => ThemeSettingProvider(
+              CustomThemeSetting<AppThemeData>(
+                activeThemes: {customTheme},
+                themes: [customTheme, customTheme2],
+              ),
+            ),
+          ),
```

Comment:
By having a first look I'd not expect this to be in the wrapper since it's something you aim to test. But I only looked at this for about 60 sec, so it might be necessary. Let me know.